buildengine/project.go

```diff
@@ -0,0 +1,153 @@
+package buildengine
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"golang.design/x/reflect"
+
+	"github.com/TBD54566975/ftl/common/moduleconfig"
+)
+
+// Project models FTL modules and external libraries and is used to manage dependencies within the build engine
+type Project interface {
```

Comment:
Add a private no-op method to this and annotate it with

```go
////sumtype:decl
```

We use [go-check-sumtype](https://github.com/alecthomas/go-check-sumtype) to statically check that "sum type" use is exhaustive.